File: HealthAppMVC/SharedDto/AppointmentDtos/AppointmentDto.cs
using System;

namespace SharedDto.AppointmentDtos
{
    public class AppointmentDto
    {
        public int AppointmentId { get; set; }

        public string PatientName { get; set; }

        public string DoctorName { get; set; }

        public DateTime ScheduledDate { get; set; }

        public string TimeSlot { get; set; }

        public string Status { get; set; }

        public string CancellationReason { get; set; }
    }
}

File: HealthApp.Angular/src/app/dtos/appointment.dto.ts
import { HealthRecordDto } from './health-record.dto';

export type AppointmentStatus =
  | 'Pending'
  | 'Confirmed'
  | 'Cancelled'
  | 'Completed';

export interface AppointmentDto {
  appointmentId: number;
  patientId: number;
  doctorId: number;
  patientName: string;
  doctorName: string;
  scheduledDate: string;
  timeSlot: string;
  status: AppointmentStatus;
  cancellationReason?: string | null;
}

export interface AppointmentCreateDto {
  patientId: number;
  doctorId: number;
  scheduledDate: string;
  timeSlot: string;
}

export interface AppointmentBookingDto {
  doctorId: number;
  scheduledDate: string;
  timeSlot: string;
}

export interface CancelAppointmentDto {
  cancellationReason: string;
}

export interface DoctorAppointmentViewDto extends AppointmentDto {
  healthRecord?: HealthRecordDto | null;
}

export interface AppointmentActionResponse {
  message: string;
}